mozdownload/scraper.py

```diff
@@ -274,8 +279,14 @@ def total_seconds(td):
 
         attempt = 0
 
-        if not os.path.isdir(self.directory):
-            os.makedirs(self.directory)
+        directory_part = self.destination
```

Comment:
I would put this into the else case to the if block below. Also please name the variable `directory` only.